app/src/main/java/edu/mit/mitmobile2/shuttles/fragment/ShuttleRouteFragment.java

```diff
@@ -188,6 +193,48 @@ protected void updateData() {
         ContentResolver.requestSync(MitMobileApplication.mAccount, MitMobileApplication.AUTHORITY, bundle);
     }
 
+    public void refreshMapInfoWindow() {
```

Comment:
Instead of creating a new interface each time, you can just have ShuttleRouteFragment implement InfoWindowAdapter, and then call setInfoWindowAdapter(this) when you want to refresh